File: shop/src/app/models/ProductInOrder.ts
import {ProductInfo} from "./productInfo";

export class ProductInOrder {
    _id: string;
    name: string;
    price: number;
    countInStock: number;
    description: string;
    image: string;
    categoryType: number;
    count: number;

    constructor(productInfo:ProductInfo, quantity = 1){
        this._id = productInfo._id;
        this.name = productInfo.name;
        this.price = productInfo.price;
        this.countInStock = productInfo.countInStock;
        this.description = productInfo.description;;
        this.image = productInfo.image;
        this.categoryType = productInfo.categoryType;
        this.count = quantity;
    }
}

File: order-service/src/main/java/com/cg/shopping/orderservice/client/ProductInOrder.java
package com.cg.shopping.orderservice.client;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class ProductInOrder {
    @NotEmpty
    private String _id;

    @NotEmpty
    private String name;

    @NotNull
    private String description;

    private String image;

    private Integer categoryType;

    @NotNull
    private BigDecimal price;

    @Min(0)
    private Integer countInStock;

    @Min(1)
    private Integer count;

}